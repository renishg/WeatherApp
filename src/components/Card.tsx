import React, {FC, PropsWithChildren} from 'react';
import {View, StyleSheet, ViewStyle} from 'react-native';
import {colors} from '../theme/colors';

type CardProps = {
  styles?: ViewStyle;
};

const Card: FC<PropsWithChildren<CardProps>> = ({
  children,
  styles: propStyles,
}) => {
  return (
    <View style={StyleSheet.flatten([styles.container, propStyles])}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 14,
    borderColor: colors.white,
    borderWidth: 2,
    paddingHorizontal: 12,
    paddingVertical: 14,
    marginVertical: 8,
    backgroundColor: colors.transparentBackground,
  },
});

export {Card};
