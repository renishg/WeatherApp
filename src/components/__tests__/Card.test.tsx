import React from 'react';
import {create} from 'react-test-renderer';
import {Card} from '../Card';

describe('Card', () => {
  it('should render with default styles', () => {
    const tree = create(<Card>{`Test`}</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with custom styles', () => {
    const tree = create(
      <Card styles={{backgroundColor: 'red'}}>{`Test`}</Card>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
