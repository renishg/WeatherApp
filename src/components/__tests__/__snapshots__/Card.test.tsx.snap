// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with custom styles 1`] = `
<View
  style={
    {
      "backgroundColor": "red",
      "borderColor": "#FFFFFF",
      "borderRadius": 14,
      "borderWidth": 2,
      "marginVertical": 8,
      "paddingHorizontal": 12,
      "paddingVertical": 14,
    }
  }
>
  Test
</View>
`;

exports[`Card should render with default styles 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(0,0,0, 0.2)",
      "borderColor": "#FFFFFF",
      "borderRadius": 14,
      "borderWidth": 2,
      "marginVertical": 8,
      "paddingHorizontal": 12,
      "paddingVertical": 14,
    }
  }
>
  Test
</View>
`;
